tasks/HSVSegmentationAndBlur.cpp

```diff
@@ -75,165 +73,156 @@ void HSVSegmentationAndBlur::updateHook()
-                size_t pos = x+(y*v_plane->width);
-                lower_lighting += v_plane->imageData[pos];
+            for(size_t y= v_plane->size().height-30; y < v_plane->size().height;y++){
+                size_t pos = x+(y*v_plane->size().width);
+                lower_lighting += v_plane->data[pos];
             }
-         }
-         double correction = (double)upper_lighting-lower_lighting;
-         correction /= num_pixel*v_plane->height;
-         printf("Correctoin factor: %f (%lu,%lu)\n",correction,upper_lighting,lower_lighting);
-         
-         for(size_t x= 0; x < v_plane->width;x++){
+        }
+        double correction = (double)upper_lighting-lower_lighting;
+        correction /= num_pixel*v_plane->size().height;
+        printf("Correctoin factor: %f (%lu,%lu)\n",correction,upper_lighting,lower_lighting);
+        
+        for(size_t x= 0; x < v_plane->size().width;x++) {
             for(size_t y= 0; y < 30 ;y++){
-                size_t pos = x+(y*v_plane->width);
-                double v = v_plane->imageData[pos] + correction*y;
+                size_t pos = x+(y*v_plane->size().width);
+                double v = v_plane->data[pos] + correction*y;
                 if(v<0)v=0;
                 if(v>2550)v=255;
-                v_plane->imageData[pos]=v;
+                v_plane->data[pos]=v;
                 
             }
-         }
+        }
 
-         frame_helper::FrameHelper::copyMatToFrame(v_plane,*phsv_v_frame);
-         hsv_v_frame.reset(phsv_v_frame);
-         _hsv_v_frame.write(hsv_v_frame);
+        frame_helper::FrameHelper::copyMatToFrame(*v_plane,*phsv_v_frame);
+        hsv_v_frame.reset(phsv_v_frame);
+        _hsv_v_frame.write(hsv_v_frame);
 
-         cvCopy(h_plane, h_plane_debug);
-         cvCopy(s_plane, s_plane_debug);
-         cvCopy(v_plane, v_plane_debug);
+        // ----- DEBUG OUTPUT -----
 
-         cvThreshold(h_plane, h_plane, _hMax, 255, CV_THRESH_TOZERO_INV);
-         cvThreshold(h_plane, h_plane, _hMin, 255, CV_THRESH_BINARY);
-         cvThreshold(s_plane, s_plane, _sMax, 255, CV_THRESH_TOZERO_INV);
-         cvThreshold(s_plane, s_plane, _sMin, 255, CV_THRESH_BINARY);
-         cvThreshold(v_plane, v_plane, _vMax, 255, CV_THRESH_TOZERO_INV);
-         cvThreshold(v_plane, v_plane, _vMin, 255, CV_THRESH_BINARY);
+        frame_helper::FrameHelper::copyMatToFrame(*h_plane,*phd_frame);
+        phd_frame->time = in_frame->time;
+        h_frame_debug.reset(phd_frame);
+        _hDebugGray.write(h_frame_debug);
+        
+        frame_helper::FrameHelper::copyMatToFrame(*s_plane,*psd_frame);
+        psd_frame->time = in_frame->time;
+        s_frame_debug.reset(psd_frame);
+        _sDebugGray.write(s_frame_debug);
+        
+        frame_helper::FrameHelper::copyMatToFrame(*v_plane,*pvd_frame);
+        pvd_frame->time = in_frame->time;
+        v_frame_debug.reset(pvd_frame);
+        _vDebugGray.write(v_frame_debug);     
```

Comment:
yes.

if you look in the original code, the original image was copied to debug image, than the original image was modified father, but the debug image was written to the ports at the end. 
to avoid the coping of the image, I just wrote the original image as debug before the father modification was done.

Reply:
ok, makes sense